internal/cmd/pause.go

```diff
@@ -45,6 +45,53 @@ func (c *Client) PackageSetPaused(
 		delete(pkg.Annotations, PauseMessageAnnotation)
 	}
 
+	return pkg, nil
+}
+
+func packageSetPausedCluster(
+	ctx context.Context, c client.Client,
+	name string, pause bool, message string,
+) (client.Object, error) {
+	pkg := &corev1alpha1.ClusterPackage{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+	}
+	if err := c.Get(ctx, client.ObjectKeyFromObject(pkg), pkg); err != nil {
+		return pkg, fmt.Errorf("getting clusterpackage object: %w", err)
+	}
+
+	pkg.Spec.Paused = pause
+	if pause {
+		if pkg.Annotations == nil {
+			pkg.Annotations = map[string]string{}
+		}
+
+		pkg.Annotations[PauseMessageAnnotation] = message
+	} else {
+		delete(pkg.Annotations, PauseMessageAnnotation)
+	}
+
+	return pkg, nil
+}
+
+func (c *Client) PackageSetPaused(
+	ctx context.Context, waiter Waiter,
+	name, namespace string, pause bool, message string,
+) error {
+	var (
+		pkg client.Object
+		err error
+	)
+	if namespace == "" {
```

Comment:
I think this + [how the distinction between ClusterPackage and Package is made in the cmd.RunE](https://github.com/pbabic-redhat/package-operator/blob/f847d1d56ed1f7e0275d2e9120fdfb90bc40d3b1/cmd/kubectl-package/pausecmd/pause.go#L47) breaks when a user calls `pause -n foo clusterpackage/bar`.

Reply:
updated proposed diff a third time because it had a typo

Reply:
Thanks for the proposal! I integrated it, the tests needed a little tweak. :grin: